algorithms/src/std_algorithms/impl/Kokkos_CopyCopyN.hpp

```diff
@@ -94,16 +96,61 @@ OutputIterator copy_impl(const std::string& label, const ExecutionSpace& ex,
 
 template <class ExecutionSpace, class InputIterator, class Size,
           class OutputIterator>
-OutputIterator copy_n_impl(const std::string& label, const ExecutionSpace& ex,
-                           InputIterator first_from, Size count,
-                           OutputIterator first_dest) {
+OutputIterator copy_n_exespace_impl(const std::string& label,
+                                    const ExecutionSpace& ex,
+                                    InputIterator first_from, Size count,
+                                    OutputIterator first_dest) {
   // checks
   Impl::static_assert_random_access_and_accessible(ex, first_from, first_dest);
   Impl::static_assert_iterators_have_matching_difference_type(first_from,
                                                               first_dest);
 
   if (count > 0) {
-    return copy_impl(label, ex, first_from, first_from + count, first_dest);
+    return copy_exespace_impl(label, ex, first_from, first_from + count,
+                              first_dest);
+  } else {
+    return first_dest;
+  }
+}
+
+//
+// team-level impl
+//
+template <class TeamHandleType, class InputIterator, class OutputIterator>
+KOKKOS_FUNCTION OutputIterator copy_team_impl(const TeamHandleType& teamHandle,
```

Comment:
Why did you not just overload the `copy_impl` and `copy_n_impl` functions?
Conscious stylistic choice or did you not realize that you could?

Reply:
are you referring to overloading leveraging the fact that the team ones don't accept the label? 
The naming was on purpose becuase if at some point we do have a label for teams, this would not cause ambiguous errors . Does that answer?

Reply:
> are you referring to overloading leveraging the fact that the team ones don't accept the label?

Yes

> The naming was on purpose becuase if at some point we do have a label for teams, this would not cause ambiguous errors .

I do not anticipate this to change.

> Does that answer?

I suppose so, in the sense it was on purpose.

Reply:
> I do not anticipate this to change.

you mean the label for team-based kernels? if so, Christian said some time ago, that "at some point" he has a plan to make all that behavior consistent across all variations. Might be long time away though 

In any case, yes, short answer is on purpose, happy to debate about it

Reply:
What would we do with the label in a nested algorithm on the device?
@crtrott ?

Reply:
btw, i asked about this before starting to do this work during one of the weekly meetings, but I could be making a mistaking recalling this , that is always possible